include/boost/geometry/util/math.hpp

```diff
@@ -772,6 +773,141 @@ inline Result rounding_cast(T const& v)
 }
 
 /*!
+\brief Evaluate the sine and cosine function with the argument in degrees
+\note The results obey exactly the elementary properties of the trigonometric
+      functions, e.g., sin 9&deg; = cos 81&deg; = &minus; sin 123456789&deg;.
+      If x = &minus;0, then \e sinx = &minus;0; this is the only case where
+      &minus;0 is returned.
+*/
+template<typename T>
+inline void sin_cos_degrees(T const& x,
+                            T & sinx,
+                            T & cosx)
+{
+    // In order to minimize round-off errors, this function exactly reduces
+    // the argument to the range [-45, 45] before converting it to radians.
+    T remainder; int quotient;
+
+    remainder = math::mod(x, T(360));
+    quotient = floor(remainder / 90 + T(0.5));
+    remainder -= 90 * quotient;
+
+    // Convert to radians.
+    remainder *= d2r<T>();
+
+    T s = sin(remainder), c = cos(remainder);
+
+    switch (unsigned(quotient) & 3U)
+    {
+        case 0U: sinx =  s; cosx =  c; break;
+        case 1U: sinx =  c; cosx = -s; break;
+        case 2U: sinx = -s; cosx = -c; break;
+        default: sinx = -c; cosx =  s; break; // case 3U
+    }
+
+    // Set sign of 0 results. -0 only produced for sin(-0).
+    if (x != 0)
+    {
+        sinx += T(0); cosx += T(0);
+    }
+}
+
+/*!
+\brief The NaN (not a number)
+*/
+template<typename T>
+inline T NaN()
+{
+#if defined(_MSC_VER)
+    return std::numeric_limits<T>::has_quiet_NaN ?
+      std::numeric_limits<T>::quiet_NaN() :
+      (std::numeric_limits<T>::max)();
+#else
+    return std::numeric_limits<T>::has_quiet_NaN ?
+      std::numeric_limits<T>::quiet_NaN() :
+      std::numeric_limits<T>::max();
+#endif
```

Comment:
Parentheses around `max` would work in all compilers `#if` is not needed.

But there is a greater problem with this code. It will not work for user-defined, non-fundamental types because for them `numeric_limits` will not be specialized. So I'd remove `NaN` function and replace it with some other mechanism.

Reply:
`math::NaN()` was previously being used when the angle was greater
than 90 deg. Since we now use the maximum angle value [-90, 90]
(see comment for "NaN() for fundamental types may return `quiet_nan`
 or `max` depending on the compiler.") this function is no longer required.